rfcs/0000-oidc-policy.md

```diff
@@ -0,0 +1,185 @@
+# `OpenIDConnectPolicy` - a metapolicy
+
+- Feature Name: oidc-policy
+- Start Date: 2025-02-05
+- RFC PR: [Kuadrant/architecture#0114](https://github.com/Kuadrant/architecture/pull/114)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+The `OpenIDConnectPolicy` aims at helping users to easily & quickly get started with the Kuadrant stack to setup an
+[OpenID Connect flow](https://openid.net/developers/how-connect-works/) without needing to initially understand how to
+use Kuadrant's `AuthPolicy` to achieve their goals. 
+
+# Motivation
+[motivation]: #motivation
+
+There are a few goals this implementation hopes to achieve and set a precedent for introducing more such APIs in
+Kuadrant:
+
+ - users looking at an example policy would be able to make mostly sense of it without needing to lookup documentation;
+ - the policy would apply sensible defaults in most cases and only require the very minimal input from the user;
+ - the idea is to implement it as _metapolicy_, which would only be consumed by a controller to produce a `AuthPolicy`
+   that would then enforce the flow;
+ - it is _not_ the goal to expose all the power of the `AuthPolicy` to the user through the new policy, but rather to let
+   users achieve most of their goals by slowly exploring defaults of the policy and eventually dig down the stack, into
+   the `AuthPolicy` to expand their use cases when they need it;
+ - explore how to, over time, grow the initial minimal API of the policy into a larger one to cover more use cases than
+   the one currently set as the goal for this RFC
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `OpenIDConnectPolicy` is a direct policy attachment that attaches to one or multiple
+[`HTTPRoute`](https://gateway-api.sigs.k8s.io/api-types/httproute/) objects. Targeted routes will require requests to
+carry an access token to be authorized through. Should the token be missing or be invalid, the requesting entity gets
+redirected to the `provider`'s `authorizationEndpoint` specified in the policy for them to identify.
+
+It is important to note here that the access token is to be negotiated stored by the client. While this may change in
+the future, currently the token is to either be provided by the appropriate HTTP header (`Authorization: Bearer`, which 
+is the default source), or by a Cookie.
```

Comment:
impl detail, but thoughts on cookie name? oidc connect novice, but unaware of a convention (as opposed to the authz: bearer header)

Reply:
other random thought: what has precedence if both present?

Reply:
`tokenSource: ` would be overridden to e.g. `cookie`. So either we make it aware of the `name` to look for (e.g. `cookie["auth"]`; or make it another config bit (e.g. `tokenSourceKey`?); and/or with a sensible default (e.g. `Bearer`?); I'm wondering if this wouldn't best be expressed as CEL ... "somehow".

wrt to precedence, it don't matter, as the user specify what "source" to use.